code/lib/preview-api/src/modules/store/csf/normalizeProjectAnnotations.ts

```diff
@@ -13,6 +13,8 @@ import { normalizeInputTypes } from './normalizeInputTypes';
 import { normalizeArrays } from './normalizeArrays';
 import { combineParameters } from '../parameters';
 
+// TODO(kasperpeulen) Consolidate this function with composeConfigs
+// As composeConfigs is the real normalizer, and always run before normalizeProjectAnnotations
```

Comment:
I discussed this with someone recently (@shilman I think?). Alternatively we could get rid of `composeConfigs` and just pass `ProjectAnnotations[]` around -- and do the composing here.

That makes sense to me as it avoids the need for both WP + Vite to call `composeConfigs` at the right time.

It definitely doesn't makes sense to have both `composeConfigs` and `normalizeProjectAnnotations`.

Also I don't think these two lines belong in there: https://github.com/storybookjs/storybook/blob/8bb129890fd02e74a5c4a5c5df9296fdcb74a2da/code/lib/preview-api/src/modules/store/csf/normalizeProjectAnnotations.ts#L43-L48

Instead, they should get added to the `ProjectAnnotations[]` somewhere.

Reply:
Right, I like that. It seems better to do as little as possible JS in the webpack/vite template files.

Those 2 lines, should move to the addon-controls preview file right?

Reply:
> Those 2 lines, should move to the addon-controls preview file right?

Possibly, let's discuss with @shilman to make sure.